<a href="https://colab.research.google.com/github/elhadidiilias3612encg-hash/ILIAS-EL-HADIDI/blob/main/Copie_de_Bienvenue_dans_Colab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

on commancant par l'importation du package de "the ucimlrepo package" pour faciliter la tâche de l'analyse

In [ ]:
pip install ucimlrepo

In [ ]:
from ucimlrepo import fetch_ucirepo

# fetch dataset
bank_marketing = fetch_ucirepo(id=222)

# data (as pandas dataframes)
X = bank_marketing.data.features
y = bank_marketing.data.targets

# metadata
print(bank_marketing.metadata)

# variable information
print(bank_marketing.variables)


{'uci_id': 222, 'name': 'Bank Marketing', 'repository_url': 'https://archive.ics.uci.edu/dataset/222/bank+marketing', 'data_url': 'https://archive.ics.uci.edu/static/public/222/data.csv', 'abstract': 'The data is related with direct marketing campaigns (phone calls) of a Portuguese banking institution. The classification goal is to predict if the client will subscribe a term deposit (variable y).', 'area': 'Business', 'tasks': ['Classification'], 'characteristics': ['Multivariate'], 'num_instances': 45211, 'num_features': 16, 'feature_types': ['Categorical', 'Integer'], 'demographics': ['Age', 'Occupation', 'Marital Status', 'Education Level'], 'target_col': ['y'], 'index_col': None, 'has_missing_values': 'yes', 'missing_values_symbol': 'NaN', 'year_of_dataset_creation': 2014, 'last_updated': 'Fri Aug 18 2023', 'dataset_doi': '10.24432/C5K306', 'creators': ['S. Moro', 'P. Rita', 'P. Cortez'], 'intro_paper': {'ID': 277, 'type': 'NATIVE', 'title': 'A data-driven approach to predict the s

on ajoute de commentaires pour clarifier et optimiser les données

This dataset is related to direct marketing campaigns (phone calls) of a Portuguese banking institution.

### **Dataset Overview:**
*   **Name:** Bank Marketing
*   **Abstract:** The primary goal is to predict if a client will subscribe to a term deposit (variable 'y').
*   **Number of Instances:** 45,211
*   **Number of Features:** 16
*   **Tasks:** Classification
*   **Missing Values:** Yes, indicated by 'NaN' (for features like `contact`, `pdays`, `poutcome`)

### **Key Variables Information:**
*   **`age`**: Numeric, client's age.
*   **`job`**: Categorical, type of job (e.g., 'admin.', 'blue-collar', 'management').
*   **`marital`**: Categorical, marital status (e.g., 'married', 'single', 'divorced').
*   **`education`**: Categorical, education level (e.g., 'secondary', 'tertiary', 'primary').
*   **`default`**: Binary, indicates if the client has credit in default ('yes'/'no').
*   **`balance`**: Numeric, average yearly balance in euros.
*   **`housing`**: Binary, indicates if the client has a housing loan ('yes'/'no').
*   **`loan`**: Binary, indicates if the client has a personal loan ('yes'/'no').
*   **`contact`**: Categorical, contact communication type (e.g., 'cellular', 'telephone', 'unknown').
*   **`day`**: Numeric, last contact day of the month.
*   **`month`**: Categorical, last contact month of year (e.g., 'jan', 'feb', ..., 'dec').
*   **`duration`**: Numeric, last contact duration in seconds. **(Important Note: This attribute highly influences the output target 'y'. If duration=0, then 'y'='no'. You might want to discard this attribute if the goal is to build a realistic predictive model, as the duration is not known before the call is performed.)**
*   **`campaign`**: Numeric, number of contacts performed during this campaign for the client.
*   **`pdays`**: Numeric, number of days that passed after the client was last contacted from a previous campaign (-1 means not previously contacted).
*   **`previous`**: Numeric, number of contacts performed before this campaign for the client.
*   **`poutcome`**: Categorical, outcome of the previous marketing campaign (e.g., 'failure', 'success', 'other', 'unknown').

### **Target Variable:**
*   **`y`**: Binary, indicates if the client subscribed to a term deposit ('yes'/'no').